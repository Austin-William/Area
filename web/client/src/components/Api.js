import React from 'react';
import axios from 'axios';
import { url } from '../global/variables';
import { NotificationManager } from 'react-notifications';
import "react-notifications/lib/notifications.css";
import "../styles/Api.css";
// import { getAuth } from "firebase/auth";
// import { initializeApp } from 'firebase/app';

function Api(props) {

    const [access_token, setAccess_token] = React.useState('');

    const handleAccessToken = (e) => {
        setAccess_token(e.target.value);
    }

    // React.useEffect(() => {
    //     axios.get(url + '/firebase').then((res) => initializeApp(res.data));
    // })
    

    const handleSubmit = (e) => {
        e.preventDefault();
        // const auth = getAuth();
        // axios.get(url + '/api/user/' + auth.currentUser.email).then(result => {
        //     const userData = result.data;
        //     console.log(userData);
        // })
        if (props.access_token === "access_token") {
            if (access_token.length > 0) {
                axios.post(url + props.link + access_token)
                    .then(res => {
                        NotificationManager.success(res.data.message + "\n\nYou successfully subscribe to " + props.name, 'Success', 3000);
                    })
                    .catch(err => {
                        console.log(err);
                    });
                let newSubscription = props.currentUserData.subscriptions;
                if (newSubscription[props.index].state === false) {
                    newSubscription[props.index].state = true;
                    axios.put(url + '/api/user/' + props.currentUserData.id, {
                        "subscriptions": newSubscription
                    });
                } else if (newSubscription[props.index].state === true) {
                    NotificationManager.error("You already subscribe to " + props.name, "Error", 3000);
                }
            }
        } else {
            let newSubscription = props.currentUserData.subscriptions;
            if (newSubscription[props.index].state === false) {
                newSubscription[props.index].state = true;
                axios.put(url + '/api/user/' + props.currentUserData.id, {
                    "subscriptions": newSubscription
                });
                NotificationManager.success("You successfully subscribe to " + props.name, "Success", 3000);
            } else if (newSubscription[props.index].state === true) {
                NotificationManager.error("You already subscribe to " + props.name, "Error", 3000);
            }
        }
    }

    return (
        <div className="Api">
            <div className="Api-container">
                <div className="Api-header">
                    <h1>
                        {props.name}
                    </h1>
                    {
                        props.access_token === "access_token" ?

                            <input type="text" placeholder="Access Token" onChange={handleAccessToken} />
                            :

                            props.access_token === "none" ?
                                <p>No Access Token Required</p>
                                :
                                <>
                                    <input type="text" placeholder="Email" />
                                    <input type="text" placeholder="Password" />
                                </>

                    }
                </div>
                <div className="Api-footer">
                    <button onClick={handleSubmit}>
                        Add
                    </button>
                </div>
            </div>
        </div>
    );
}

export default Api;