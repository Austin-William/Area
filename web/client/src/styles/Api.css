.Api {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: #fafafa;
    padding: 20px;
    margin: 2em;
    border-radius: 5px;
    max-width: 20em;
}

.Api-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.Api-header {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-bottom: 1em;
}

.Api-header input {
    width: 100%;
    padding: 10px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
    margin-bottom: 5px;
    transition: all 0.5s ease-in-out;
}

.Api-header input:hover {
    border: 1px solid #777;
}

.Api-header input:focus {
    outline: none;
    border: 1px solid #777;
}

.Api-footer {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.Api-footer button {
    width: 100%;
    padding: 10px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
    margin-bottom: 5px;
    transition: all 0.5s ease-in-out;
}

.Api-footer button:hover {
    border: 1px solid #777;
}