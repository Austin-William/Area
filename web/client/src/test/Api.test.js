import React from "react";
import renderer from "react-test-renderer";
import Api from "../components/Api";

test("Components : Api renders correctly", () => {
  const tree = renderer.create(<Api />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Components : Api renders correctly", () => {
  const tree = renderer.create(<Api link="https://areactionback.herokuapp.com/" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Components : Api renders correctly", () => {
  const tree = renderer.create(<Api name="test" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Components : Api renders correctly", () => {
  const tree = renderer.create(<Api access_token="access_token_random" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Components : Api renders correctly", () => {
  const tree = renderer.create(<Api index={5} />).toJSON();
  expect(tree).toMatchSnapshot();
});