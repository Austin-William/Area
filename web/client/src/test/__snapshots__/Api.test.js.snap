// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Api renders correctly 1`] = `
<div
  className="Api"
>
  <div
    className="Api-container"
  >
    <div
      className="Api-header"
    >
      <h1 />
      <input
        placeholder="Email"
        type="text"
      />
      <input
        placeholder="Password"
        type="text"
      />
    </div>
    <div
      className="Api-footer"
    >
      <button
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
</div>
`;

exports[`Components : Api renders correctly 2`] = `
<div
  className="Api"
>
  <div
    className="Api-container"
  >
    <div
      className="Api-header"
    >
      <h1 />
      <input
        placeholder="Email"
        type="text"
      />
      <input
        placeholder="Password"
        type="text"
      />
    </div>
    <div
      className="Api-footer"
    >
      <button
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
</div>
`;

exports[`Components : Api renders correctly 3`] = `
<div
  className="Api"
>
  <div
    className="Api-container"
  >
    <div
      className="Api-header"
    >
      <h1>
        test
      </h1>
      <input
        placeholder="Email"
        type="text"
      />
      <input
        placeholder="Password"
        type="text"
      />
    </div>
    <div
      className="Api-footer"
    >
      <button
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
</div>
`;

exports[`Components : Api renders correctly 4`] = `
<div
  className="Api"
>
  <div
    className="Api-container"
  >
    <div
      className="Api-header"
    >
      <h1 />
      <input
        placeholder="Email"
        type="text"
      />
      <input
        placeholder="Password"
        type="text"
      />
    </div>
    <div
      className="Api-footer"
    >
      <button
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
</div>
`;

exports[`Components : Api renders correctly 5`] = `
<div
  className="Api"
>
  <div
    className="Api-container"
  >
    <div
      className="Api-header"
    >
      <h1 />
      <input
        placeholder="Email"
        type="text"
      />
      <input
        placeholder="Password"
        type="text"
      />
    </div>
    <div
      className="Api-footer"
    >
      <button
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
</div>
`;
